tools/c-parser/Simpl/ex/Quicksort.thy

```diff
@@ -217,65 +167,55 @@ apply (hoare_rule anno =
        \<acute>le :== CALL append(\<acute>le,\<acute>p);;
        \<acute>p :== \<acute>le
   FI" in HoarePartial.annotateI)
-apply vcg
-apply   fastforce
-apply  clarsimp
-apply  (rule conjI)
-apply   clarify
-apply   (rule conjI)
-apply    (rule_tac x="tl#les" in exI)
-apply    simp
-apply    (rule_tac x="grs" in exI)
-apply    simp
-apply    (rule_tac x="ps" in exI)
-apply    simp
-apply    (erule perm.trans)
-apply    simp
-apply    (simp add: perm_app_Cons_simps)
-apply   (simp add: perm_set_eq)
-apply  clarify
-apply  (rule conjI)
-apply   (rule_tac x="les" in exI)
-apply   simp
-apply   (rule_tac x="tl#grs" in exI)
-apply   simp
-apply   (rule_tac x="ps" in exI)
-apply   simp
-apply   (erule perm.trans)
-apply   simp
-apply   (simp add: perm_app_Cons_simps)
-apply  (simp add: perm_set_eq)
-apply clarsimp
-apply (rule_tac ?x=grs in exI)
-apply (rule conjI)
-apply  (erule heap_eq_ListI1)
-apply  clarify
-apply  (erule_tac x=x in allE)back
-apply  blast
-apply clarsimp
-apply (rule_tac x="sortedPs" in exI)
-apply (rule conjI)
-apply  (erule heap_eq_ListI1)
-apply  (clarsimp)
-apply  (erule_tac x=x in allE) back back
-apply  (fast dest!: perm_set_eq)
-apply (rule_tac x="p#sortedPsa" in exI)
-apply (rule conjI)
-apply  (fastforce dest!: perm_set_eq)
-apply (rule conjI)
-apply  (force dest!: perm_set_eq)
-apply clarsimp
-apply (rule conjI)
-apply  (fastforce dest!: perm_set_eq)
-apply (rule conjI)
-apply  (fastforce dest!: perm_set_eq)
-apply (rule conjI)
-apply  (erule perm.trans)
-apply  (simp add:  perm_app_Cons_simps list_all_iff)
-apply  (fastforce intro!: perm_append_blocks)
-apply clarsimp
-apply (erule_tac x=x in allE)+
-apply (force dest!: perm_set_eq)
-done
+  apply vcg
+    apply fastforce
+   apply clarsimp
+   apply (rule conjI)
+    apply clarify
+    apply (rule conjI)
+     apply (rule_tac x="tl#les" in exI)
+     apply simp
+     apply (rule_tac x="grs" in exI)
+     apply simp
+     apply (rule_tac x="ps" in exI)
+     apply simp
+    apply (metis insertCI set_mset_add_mset_insert set_mset_mset)
+   apply clarify
+   apply (rule conjI)
+    apply (rule_tac x="les" in exI)
+    apply simp
+    apply (rule_tac x="tl#grs" in exI)
+    apply simp
+    apply (rule_tac x="ps" in exI)
+    apply simp
+   apply (metis insertCI set_mset_add_mset_insert set_mset_mset)
+  apply clarsimp
+  apply (rule_tac ?x=grs in exI)
+  apply (rule conjI)
+  apply (erule heap_eq_ListI1)
+   apply clarify
+   apply (erule_tac x=x in allE) back
```

Comment:
are these style failures due to `back` ancient Harvey code?

Reply:
This is `Simpl/ex/Quicksort`, so a demo (which seems to have changed in the AFP import). No Harvey connection this time :-)